src/cascadia/TerminalApp/CommandPalette.h

```diff
@@ -7,6 +7,27 @@
 #include "CommandPalette.g.h"
 #include "AppCommandlineArgs.h"
 
+struct hash_winrt_object_as_pointer
+{
+    size_t operator()(const ::winrt::Windows::Foundation::IUnknown& value) const noexcept
+    {
+        void* const abi_value = winrt::get_abi(value.try_as<::winrt::Windows::Foundation::IUnknown>());
+        return std::hash<void*>{}(abi_value);
+    }
```

Comment:
The extra `try_as` seems redundant? It might make sense to do
```cpp
size_t operator()(const auto& value) const noexcept
{
    return std::hash<void*>{}(winrt::get_abi(value));
}
```

or alternatively
```cpp
size_t operator()(const auto& value) const noexcept
{
    return til::hash(winrt::get_abi(value));
}
```

Due to the novel design choices in `std::unordered_map` I suspect there's not a particularly big difference between the two though, so I don't think it matters what you pick.

Reply:
Eh. I copied this directly out of `winrt/base.h`.